test/e2e/logicModules/KPIRewarderLifecycle.t.sol

```diff
@@ -0,0 +1,527 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {E2ETest} from "test/e2e/E2ETest.sol";
+import "forge-std/console.sol";
+
+//Internal Dependencies
+import {ModuleTest, IOrchestrator} from "test/modules/ModuleTest.sol";
+import {IModule, ERC165} from "src/modules/base/Module.sol";
+import {IOrchestratorFactory} from "src/factories/OrchestratorFactory.sol";
+import {AuthorizerMock} from "test/utils/mocks/modules/AuthorizerMock.sol";
+
+// External Libraries
+import {Clones} from "@oz/proxy/Clones.sol";
+
+import {RebasingFundingManager} from
+    "src/modules/fundingManager/RebasingFundingManager.sol";
+
+import {
+    SimplePaymentProcessor,
+    IPaymentProcessor
+} from "src/modules/paymentProcessor/SimplePaymentProcessor.sol";
+
+import {
+    KPIRewarder,
+    IKPIRewarder,
+    IOptimisticOracleIntegrator,
+    OptimisticOracleV3Interface,
+    IStakingManager,
+    IERC20PaymentClient
+} from "src/modules/logicModule/KPIRewarder.sol";
+
+// Mocks
+import {ERC20Mock} from "test/utils/mocks/ERC20Mock.sol";
+import {ERC20} from "@oz/token/ERC20/ERC20.sol";
+import {IERC20Metadata} from "@oz/token/ERC20/extensions/IERC20Metadata.sol";
+
+/*
+Fork testing necessary:
+
+forge test --match-contract KPIRewarderLifecycle -vvv --fork-url $SEPOLIA_RPC_URL
```

Comment:
How exactly are you doing this in the CICD?
Can you isolate a single test to be forked tested or are all tests running now via fork?

Reply:
The code in line 123 is the one that sets up the fork, so all other tests run as usual. I updated the comment though, since it was outdated